packages/paste-website/src/utils/__tests__/DataUtils.spec.ts

```diff
@@ -0,0 +1,136 @@
+import {getNormalizedNavigationData} from '../DataUtils';
+
+const mockNavData = [
+  {
+    Feature: 'Account Switcher',
+    status: 'production',
+    Figma: 'yes',
+    'Design committee review': 'yes',
+    'Engineer committee review': 'yes',
+    Documentation: true,
+    'Component Category': 'component',
+    Code: 'yes',
+  },
+  {
+    Feature: 'Alert',
+    status: 'production',
+    Figma: 'yes',
+    'Design committee review': 'yes',
+    'Engineer committee review': 'yes',
+    Documentation: true,
+    'Component Category': 'component',
+    Code: 'yes',
+  },
+  {
+    Feature: 'Button vs Anchor',
+    status: 'production',
+    Figma: 'yes',
+    'Design committee review': 'yes',
+    'Engineer committee review': 'yes',
+    Documentation: true,
+    'Component Category': 'pattern',
+    Code: 'not applicable',
+    'Product suitability': ['Console', 'Flex', 'SendGrid'],
+  },
+  {
+    Feature: 'Charting guidelines',
+    status: 'production',
+    Documentation: true,
+    'Component Category': 'Foundation',
+    Code: 'yes',
+  },
+  {
+    Feature: 'Combobox Primitive',
+    status: 'production',
+    Figma: 'not applicable',
+    'Design committee review': 'not applicable',
+    'Engineer committee review': 'not applicable',
+    Documentation: true,
+    'Component Category': 'primitive',
+    Code: 'yes',
+  },
+  {
+    Feature: 'Complex filters',
+    status: 'in development',
+    'Component Category': 'pattern',
+  },
+  {
+    Feature: 'Grid',
+    status: 'production',
+    Figma: 'yes',
+    'Design committee review': 'not applicable',
+    'Engineer committee review': 'not applicable',
+    Documentation: true,
+    'Component Category': 'layout',
+    Code: 'yes',
+  },
+  {
+    Feature: 'Meter',
+    status: 'in development',
+    'Component Category': 'component',
+    Code: 'yes',
+  },
+];
+
+describe('getNormalizedNavigationData', () => {
+  test('should return an empty object if the input data is empty', () => {
+    const data: [] = [];
+    const result = getNormalizedNavigationData(data);
+    expect(result).toEqual({allPasteComponent: [], allPasteLayout: [], allPastePrimitive: [], allPastePattern: []});
+  });
+
+  test('should flatten and mutate the input data correctly for allPasteComponent', () => {
+    const result = getNormalizedNavigationData(mockNavData);
+    expect(result).toEqual({
```

Comment:
suggestion: instead of toEqual, these type of tests are much easier with snapshots (eg. toMatchSnaptshot())